tests/pytests/integration/states/test_idem.py

```diff
@@ -5,13 +5,13 @@
 from contextlib import contextmanager
 
 import pytest
+import salt.state
 import salt.utils.idem as idem
 import salt.utils.path
+import tests.support.sminion
 
-pytestmark = pytest.mark.skipif(not idem.HAS_POP[0], reason=idem.HAS_POP[1])
 
-
-@pytest.mark.skipif(not salt.utils.path.which("idem"), reason="idem is not installed")
+@pytest.mark.skipif(not idem.HAS_POP[0], reason="idem is not installed")
```

Comment:
Why are you preferring the decorator over the `pytestmark`?

Reply:
The pytestmark line was giving me  pylint errors

Reply:
fyi, `pytestmark` applies to all test functions, and salt's lint won't complain.